protoarray/src/main/java/tech/pegasys/teku/protoarray/ForkChoiceStrategy.java

```diff
@@ -451,4 +453,15 @@ public void onExecutionPayloadResult(
       protoArrayLock.writeLock().unlock();
     }
   }
+
+  public static class ForkChoiceStateAndIsHeadOptimistic {
+    public final ForkChoiceState forkChoiceState;
+    public final boolean isHeadOptimistic;
```

Comment:
I think I'd just add `isHeadOptimistic` to `ForkChoiceState` rather than adding a wrapper class.  We have deliberately separated the internal `ForkChoiceState` class we use from the `ForkChoiceStateV1` that defines how it serialises so can add extra information that's useful to us without it affecting the execution engine API.